src/components/common/Dialogs/CreateDomainDialog/CreateDomainDialog.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+// import { Id } from '@colony/colony-js';
+import { string, object, boolean, number, InferType } from 'yup';
+
+import Dialog, { DialogProps, ActionDialogProps } from '~shared/Dialog';
+import { ActionHookForm as Form } from '~shared/Fields';
+
+import { Color } from '~types';
+import { ActionTypes } from '~redux/index';
+import { WizardDialogType } from '~hooks'; // useEnabledExtensions
+import { pipe, withMeta, mapPayload } from '~utils/actions';
+
+import CreateDomainDialogForm from './CreateDomainDialogForm';
+import { getCreateDomainDialogPayload } from './helpers';
+
+type Props = DialogProps & WizardDialogType<object> & ActionDialogProps;
+
+const displayName = 'common.CreateDomainDialog';
+
+const validationSchema = object()
+  .shape({
+    forceAction: boolean().defined(),
+    teamName: string().max(20).required(),
+    domainColor: number().notRequired(),
+    domainPurpose: string().max(90).notRequired(),
+    annotationMessage: string().max(4000).notRequired(),
+  })
+  .defined();
+
+type FormValues = InferType<typeof validationSchema>;
+
+const CreateDomainDialog = ({
+  callStep,
+  prevStep,
+  cancel,
+  close,
+  colony,
+}: Props) => {
+  const [isForce, setIsForce] = useState(false);
+  const navigate = useNavigate();
+
+  // const { isVotingExtensionEnabled } = useEnabledExtensions({
+  //   colonyAddress: colony.colonyAddress,
+  // });
+
+  const getFormAction = (actionType: 'SUBMIT' | 'ERROR' | 'SUCCESS') => {
+    const actionEnd = actionType === 'SUBMIT' ? '' : `_${actionType}`;
+
+    return !isForce // && isVotingExtensionEnabled
+      ? ActionTypes[`MOTION_DOMAIN_CREATE_EDIT${actionEnd}`]
+      : ActionTypes[`ACTION_DOMAIN_CREATE${actionEnd}`];
+  };
+
+  const transform = pipe(
+    mapPayload((payload) => getCreateDomainDialogPayload(colony, payload)),
+    withMeta({ navigate }),
+  );
+
+  return (
+    <Form<FormValues>
+      defaultValues={{
+        forceAction: false,
+        teamName: '',
+        domainColor: Color.LightPink,
+        domainPurpose: '',
+        annotationMessage: '',
+        // motionDomainId: Id.RootDomain,
+      }}
+      submit={getFormAction('SUBMIT')}
```

Comment:
This `getFormAction` logic is being duplicated in every form. 

An alternative is to move the part concerned with the action type suffix into a util, which gets called directly inside the ActionForm

``` js
export const getFormAction = (
  action: ActionTypes,
  actionType: 'ERROR' | 'SUCCESS',
) => `${action}_${actionType}`;
```

then 

``` js
const ActionHookForm = <V extends Record<string, any>>({
  actionType,
  onSuccess,
  onError,
  onSubmitError,
  transform,
  ...props
}: Props<V>) => {
  const submit = actionType;
  const error = getFormAction(actionType, 'ERROR');
  const success = getFormAction(actionType, 'SUCCESS');
 ...
 }
```

Then you only need to pass in the actionType, the logic for which can be kept in the parent component. E.g.

``` tsx 
const actionType = !isForce /*&& isVotingExtensionEnabled*/
      ? ActionTypes.MOTION_DOMAIN_CREATE_EDIT
      : ActionTypes.ACTION_DOMAIN_CREATE

```

Reply:
@ArmandoGraterol Adding to this for completeness, Raul mentioned in one of my prs that this should also include a way to pass in the states individually, but that they have default values assigned

https://github.com/JoinColony/colonyCDapp/pull/261#discussion_r1107359898

Reply:
Thank you for the update :+1: